### ISSUE TITLE
[release/1.4.1] scalability envelope test_distributed broken since 1.3

### ISSUE THREAD

**Issue by krfricke** [ray-project/ray#16612]
### What is the problem?
The log output is filled with these:

```
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,592 E 20503 20503] logging.cc:441: *** Aborted at 1624357040 (unix time) try "date -d @1624357040" if you are using GNU date ***
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,592 E 20503 20503] logging.cc:441: PC: @                0x0 (unknown)
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,593 E 20503 20503] logging.cc:441: *** SIGABRT (@0x3e800005017) received by PID 20503 (TID 0x7f37eafd8840) from PID 20503; stack trace: ***
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,593 E 20503 20503] logging.cc:441:     @     0x560add24428f google::(anonymous namespace)::FailureSignalHandler()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,593 E 20503 20503] logging.cc:441:     @     0x7f37eabd18a0 (unknown)
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,593 E 20503 20503] logging.cc:441:     @     0x7f37e9cc5f47 gsignal
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,593 E 20503 20503] logging.cc:441:     @     0x7f37e9cc78b1 abort
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,594 E 20503 20503] logging.cc:441:     @     0x560add2303fe ray::SpdLogMessage::Flush()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,594 E 20503 20503] logging.cc:441:     @     0x560add2304cd ray::RayLog::~RayLog()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,595 E 20503 20503] logging.cc:441:     @     0x560add1e512f ray::rpc::GrpcServer::Run()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,595 E 20503 20503] logging.cc:441:     @     0x560adcee387c ray::ObjectManager::StartRpcService()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,595 E 20503 20503] logging.cc:441:     @     0x560adcef9ef8 ray::ObjectManager::ObjectManager()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,596 E 20503 20503] logging.cc:441:     @     0x560adce825d7 ray::raylet::NodeManager::NodeManager()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,596 E 20503 20503] logging.cc:441:     @     0x560adce20725 ray::raylet::Raylet::Raylet()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,596 E 20503 20503] logging.cc:441:     @     0x560adcde8ae0 _ZZ4mainENKUlN3ray6StatusERKN5boost8optionalISsEEE_clES0_S5_
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,596 E 20503 20503] logging.cc:441:     @     0x560adcde997b _ZNSt17_Function_handlerIFvN3ray6StatusERKN5boost8optionalISsEEEZ4mainEUlS1_S6_E_E9_M_invokeERKSt9_Any_dataOS1_S6_
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,596 E 20503 20503] logging.cc:441:     @     0x560adcfbe11d _ZNSt17_Function_handlerIFvRKN3ray6StatusERKNS0_3rpc22GetInternalConfigReplyEEZNS0_3gcs28ServiceBasedNodeInfoAccessor22AsyncGetInternalConfigERKSt8functionIFvS1_RKN5boost8optionalISsEEEEEUlS3_S7_E_E9_M_invokeERKSt9_Any_dataS3_S7_
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,597 E 20503 20503] logging.cc:441:     @     0x560adcf6dea1 _ZNSt17_Function_handlerIFvRKN3ray6StatusERKNS0_3rpc22GetInternalConfigReplyEEZNS4_12GcsRpcClient17GetInternalConfigERKNS4_24GetInternalConfigRequestERKSt8functionIS8_EEUlS3_S7_E_E9_M_invokeERKSt9_Any_dataS3_S7_
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,597 E 20503 20503] logging.cc:441:     @     0x560adcf74372 ray::rpc::ClientCallImpl<>::OnReplyReceived()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,598 E 20503 20503] logging.cc:441:     @     0x560adce52b9b _ZNSt17_Function_handlerIFvvEZN3ray3rpc17ClientCallManager29PollEventsFromCompletionQueueEiEUlvE_E9_M_invokeERKSt9_Any_data
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,598 E 20503 20503] logging.cc:441:     @     0x560add1e8478 boost::asio::detail::completion_handler<>::do_complete()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,599 E 20503 20503] logging.cc:441:     @     0x560add5b97d1 boost::asio::detail::scheduler::do_run_one()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,600 E 20503 20503] logging.cc:441:     @     0x560add5b9901 boost::asio::detail::scheduler::run()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,601 E 20503 20503] logging.cc:441:     @     0x560add5bbb00 boost::asio::io_context::run()
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,601 E 20503 20503] logging.cc:441:     @     0x560adcdb21a0 main
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,601 E 20503 20503] logging.cc:441:     @     0x7f37e9ca8b97 __libc_start_main
(raylet, ip=172.31.16.191) [2021-06-22 10:17:20,602 E 20503 20503] logging.cc:441:     @     0x560adcdd19c5 (unknown)
(raylet, ip=172.31.16.191) E0622 10:12:52.244828773   19159 server_chttp2.cc:40]        {"created":"@1624356772.244770874","description":"No address added out of total 1 resolved","file":"external/com_github_grpc_grpc/src/core/ext/transport/chttp2/server/chttp2_server.cc","file_line":394,"referenced_errors":[{"created":"@1624356772.244769414","description":"Failed to add any wildcard listeners","file":"external/com_github_grpc_grpc/src/core/lib/iomgr/tcp_server_posix.cc","file_line":340,"referenced_errors":[{"created":"@1624356772.244756446","description":"Unable to configure socket","fd":27,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/tcp_server_utils_posix_common.cc","file_line":207,"referenced_errors":[{"created":"@1624356772.244748951","description":"Address already in use","errno":98,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/tcp_server_utils_posix_common.cc","file_line":181,"os_error":"Address already in use","syscall":"bind"}]},{"created":"@1624356772.244769010","description":"Unable to configure socket","fd":27,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/tcp_server_utils_posix_common.cc","file_line":207,"referenced_errors":[{"created":"@1624356772.244766816","description":"Address already in use","errno":98,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/tcp_server_utils_posix_common.cc","file_line":181,"os_error":"Address already in use","syscall":"bind"}]}]}]}
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,244 C 19159 19159] grpc_server.cc:61:  Check failed: server_ Failed to start the grpc server. The specified port is 8076. This means that Ray's core components will not be able to function correctly. If the server startup error message is `Address already in use`, it indicates the server fails to start because the port is already used by other processes (such as --node-manager-port, --object-manager-port, --gcs-server-port, and ports between --min-worker-port, --max-worker-port). Try running lsof -i :8076 to check if there are other processes listening to the port.
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,244 E 19159 19159] logging.cc:441: *** Aborted at 1624356772 (unix time) try "date -d @1624356772" if you are using GNU date ***
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,245 E 19159 19159] logging.cc:441: PC: @                0x0 (unknown)
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,245 E 19159 19159] logging.cc:441: *** SIGABRT (@0x3e800004ad7) received by PID 19159 (TID 0x7f4317da3840) from PID 19159; stack trace: ***
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,245 E 19159 19159] logging.cc:441:     @     0x559ce75d528f google::(anonymous namespace)::FailureSignalHandler()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,245 E 19159 19159] logging.cc:441:     @     0x7f431799c8a0 (unknown)
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,245 E 19159 19159] logging.cc:441:     @     0x7f4316a90f47 gsignal
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,245 E 19159 19159] logging.cc:441:     @     0x7f4316a928b1 abort
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,246 E 19159 19159] logging.cc:441:     @     0x559ce75c13fe ray::SpdLogMessage::Flush()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,246 E 19159 19159] logging.cc:441:     @     0x559ce75c14cd ray::RayLog::~RayLog()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,247 E 19159 19159] logging.cc:441:     @     0x559ce757612f ray::rpc::GrpcServer::Run()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,247 E 19159 19159] logging.cc:441:     @     0x559ce727487c ray::ObjectManager::StartRpcService()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,248 E 19159 19159] logging.cc:441:     @     0x559ce728aef8 ray::ObjectManager::ObjectManager()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,248 E 19159 19159] logging.cc:441:     @     0x559ce72135d7 ray::raylet::NodeManager::NodeManager()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,248 E 19159 19159] logging.cc:441:     @     0x559ce71b1725 ray::raylet::Raylet::Raylet()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,248 E 19159 19159] logging.cc:441:     @     0x559ce7179ae0 _ZZ4mainENKUlN3ray6StatusERKN5boost8optionalISsEEE_clES0_S5_
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,248 E 19159 19159] logging.cc:441:     @     0x559ce717a97b _ZNSt17_Function_handlerIFvN3ray6StatusERKN5boost8optionalISsEEEZ4mainEUlS1_S6_E_E9_M_invokeERKSt9_Any_dataOS1_S6_
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,248 E 19159 19159] logging.cc:441:     @     0x559ce734f11d _ZNSt17_Function_handlerIFvRKN3ray6StatusERKNS0_3rpc22GetInternalConfigReplyEEZNS0_3gcs28ServiceBasedNodeInfoAccessor22AsyncGetInternalConfigERKSt8functionIFvS1_RKN5boost8optionalISsEEEEEUlS3_S7_E_E9_M_invokeERKSt9_Any_dataS3_S7_
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,249 E 19159 19159] logging.cc:441:     @     0x559ce72feea1 _ZNSt17_Function_handlerIFvRKN3ray6StatusERKNS0_3rpc22GetInternalConfigReplyEEZNS4_12GcsRpcClient17GetInternalConfigERKNS4_24GetInternalConfigRequestERKSt8functionIS8_EEUlS3_S7_E_E9_M_invokeERKSt9_Any_dataS3_S7_
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,249 E 19159 19159] logging.cc:441:     @     0x559ce7305372 ray::rpc::ClientCallImpl<>::OnReplyReceived()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,250 E 19159 19159] logging.cc:441:     @     0x559ce71e3b9b _ZNSt17_Function_handlerIFvvEZN3ray3rpc17ClientCallManager29PollEventsFromCompletionQueueEiEUlvE_E9_M_invokeERKSt9_Any_data
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,251 E 19159 19159] logging.cc:441:     @     0x559ce7579478 boost::asio::detail::completion_handler<>::do_complete()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,252 E 19159 19159] logging.cc:441:     @     0x559ce794a7d1 boost::asio::detail::scheduler::do_run_one()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,253 E 19159 19159] logging.cc:441:     @     0x559ce794a901 boost::asio::detail::scheduler::run()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,253 E 19159 19159] logging.cc:441:     @     0x559ce794cb00 boost::asio::io_context::run()
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,253 E 19159 19159] logging.cc:441:     @     0x559ce71431a0 main
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,254 E 19159 19159] logging.cc:441:     @     0x7f4316a73b97 __libc_start_main
(raylet, ip=172.31.16.191) [2021-06-22 10:12:52,255 E 19159 19159] logging.cc:441:     @     0x559ce71629c5 (unknown)
(autoscaler +9m35s) Resized to 4096 CPUs.
```

**Comment by krfricke** [comment]
Might be related to an environment setup problem on my end (ray version mismatch).

However, now I'm running into this:

```
t]2021-06-22 10:44:38,688	WARNING worker.py:1119 -- The actor or task with ID ffffffffffffffff22385b8d83629c755a08adf502000000 cannot be scheduled right now. It requires {CPU: 0.250000} for placement, but this node only has remaining {57.250000/64.000000 CPU, 179.200000 GiB/179.200000 GiB memory, 74.458815 GiB/74.458815 GiB object_store_memory, 1.000000/1.000000 node, 1.000000/1.000000 node:172.31.20.229}
. In total there are 0 pending tasks and 4481 pending actors on this node. This is likely due to all cluster resources being claimed by actors. To resolve the issue, consider creating fewer actors or increase the resources available to this Ray cluster. You can ignore this message if this Ray cluster is expected to auto-scale or if you specified a runtime_env for this task or actor because it takes time to install.
```

After which the script hangs forever.

**Comment by krfricke** [comment]
cc @rkooo567 @wuisawesome

**Comment by mwtian** [comment]
> Might be related to an environment setup problem on my end (ray version mismatch).
> 
> However, now I'm running into this:
> 
> ```
> t]2021-06-22 10:44:38,688	WARNING worker.py:1119 -- The actor or task with ID ffffffffffffffff22385b8d83629c755a08adf502000000 cannot be scheduled right now. It requires {CPU: 0.250000} for placement, but this node only has remaining {57.250000/64.000000 CPU, 179.200000 GiB/179.200000 GiB memory, 74.458815 GiB/74.458815 GiB object_store_memory, 1.000000/1.000000 node, 1.000000/1.000000 node:172.31.20.229}
> . In total there are 0 pending tasks and 4481 pending actors on this node. This is likely due to all cluster resources being claimed by actors. To resolve the issue, consider creating fewer actors or increase the resources available to this Ray cluster. You can ignore this message if this Ray cluster is expected to auto-scale or if you specified a runtime_env for this task or actor because it takes time to install.
> ```
> 
> After which the script hangs forever.

I have encountered a very similar error (unable to schedule even though with seemingly enough CPU), while running `release/xgboost_tests/workloads/train_small.py` in client mode with nightly Ray. It does not always happen.

**Comment by DmitriGekhtman** [comment]
@kai Which Ray versions might be mismatching (mismatch between which Ray environments/components)?

**Comment by wuisawesome** [comment]
Can we try rerunning this? if it only failed on 1 node, it might just be unlucky random number generation

**Comment by krfricke** [comment]
I ran it like 4 times with cluster restarts in between. Has anything changed since I tried last time?

Env mismatch has been fixed. It was just a mismatch between my local laptop environment and the cluster env.

**Comment by wuisawesome** [comment]
I ran it a few times too. Got past the original error, but it did break still (hanging on actor scheduling)

**Comment by rkooo567** [comment]
I think we should bisect this?

**Comment by rkooo567** [comment]
Can we re-run this in 1.4?

**Comment by DmitriGekhtman** [comment]
running in 1.4.0

**Comment by DmitriGekhtman** [comment]
Timeout error

[INFO 2021-06-23 19:45:33,524] e2e.py: 780  ... still waiting for command to finish (2700 seconds) ...
[ERROR 2021-06-23 19:45:40,821] e2e.py: 1268  Command returned non-success status: 1
Traceback (most recent call last):
  File "e2e.py", line 1255, in _run
    kick_off_only=kick_off_only)
  File "e2e.py", line 792, in run_session_command
    f"Command returned non-success status: {status_code}")
RuntimeError: Command returned non-success status: 1
[INFO 2021-06-23 19:45:40,870] e2e.py: 1490  Final check if everything worked.
[INFO 2021-06-23 19:45:40,877] e2e.py: 1496  Final results: {'status': 'timeout', 'last_logs': 'Command returned non-success status: 1'}
Traceback (most recent call last):
  File "e2e.py", line 1675, in <module>
    report=not args.no_report,
  File "e2e.py", line 1599, in run_test
    raise RuntimeError(last_logs)
RuntimeError: Command returned non-success status: 1

**Comment by DmitriGekhtman** [comment]
In the command history
`RAY_ADDRESS=auto TEST_OUTPUT_JSON=/tmp/release_test_out.json TEST_STATE_JSON=/tmp/release_test_state.json IS_SMOKE_TEST=0 python distributed/test_distributed.py`
produced
`Traceback (most recent call last):
  File "distributed/test_distributed.py", line 190, in <module>
    test_nodes()
  File "distributed/test_distributed.py", line 38, in test_nodes
    test_max_running_tasks()
  File "distributed/test_distributed.py", line 87, in test_max_running_tasks
    assert max_cpus - min_cpus_available > 1900, err_str
AssertionError: Only 1871.0/4836.0 cpus used.`

**Comment by wuisawesome** [comment]
that error looks like the 250 node scheduling was just slow which seems like just standard flakiness.

**Comment by DmitriGekhtman** [comment]
i got the same error the second time around.
running once more

**Comment by DmitriGekhtman** [comment]
no luck, same thing third time

**Comment by DmitriGekhtman** [comment]
People are probably helping with  Ray summit tutorials and stuff right now, but,
help?

**Comment by rkooo567** [comment]
Hey can you summarize what has happened so far?

**Comment by DmitriGekhtman** [comment]
Consistent (three times) failure of 250 node scheduling when running against Ray 1.4.0 -- this likely has little to do with Ray itself.

Details:
I ran the test against Ray 1.4.0 three times consecutively and each time got a timeout error displayed in terminal
```
[INFO 2021-06-23 19:45:33,524] e2e.py: 780 ... still waiting for command to finish (2700 seconds) ... [ERROR 2021-06-23 19:45:40,821] e2e.py: 1268 Command returned non-success status: 1 Traceback (most recent call last): File "e2e.py", line 1255, in _run kick_off_only=kick_off_only) File "e2e.py", line 792, in run_session_command f"Command returned non-success status: {status_code}") RuntimeError: Command returned non-success status: 1 [INFO 2021-06-23 19:45:40,870] e2e.py: 1490 Final check if everything worked. [INFO 2021-06-23 19:45:40,877] e2e.py: 1496 Final results: {'status': 'timeout', 'last_logs': 'Command returned non-success status: 1'} Traceback (most recent call last): File "e2e.py", line 1675, in <module> report=not args.no_report, File "e2e.py", line 1599, in run_test raise RuntimeError(last_logs) RuntimeError: Command returned non-success status: 1
```
and in command history for 
```RAY_ADDRESS=auto TEST_OUTPUT_JSON=/tmp/release_test_out.json TEST_STATE_JSON=/tmp/release_test_state.json IS_SMOKE_TEST=0 python distributed/test_distributed.py``
I see
```
Traceback (most recent call last): File "distributed/test_distributed.py", line 190, in <module> test_nodes() File "distributed/test_distributed.py", line 38, in test_nodes test_max_running_tasks() File "distributed/test_distributed.py", line 87, in test_max_running_tasks assert max_cpus - min_cpus_available > 1900, err_str AssertionError: Only 1871.0/4836.0 cpus used.
```

**Comment by rkooo567** [comment]
@ericl Given this issue exists in 1.4, why don't we unblock it and investigate it later? i think we should at least bisect between 1.3 & 1.4 or fix the flakiness of tests (which we should eventually do because the nightly dashboard will fail by this)

**Comment by DmitriGekhtman** [comment]
Failed with **Ray 1.3.0**
Running once more

Command executed
```
python e2e.py --test-config=~/code/ray/benchmarks/benchmark_tests.yaml --test-name=distributed --ray-wheels=https://ray-wheels.s3.us-west-2.amazonaws.com/releases/1.3.0/1231f35a773e5c20e13a883373e63e20f338dc3c/ray-1.3.0-cp37-cp37m-manylinux2014_x86_64.wh
```
Confirming that the commit hash is the right one:
https://github.com/ray-project/ray/tree/releases/1.3.0

Traceback from terminal
```[INFO 2021-06-25 11:08:30,465] e2e.py: 780  ... still waiting for command to finish (1830 seconds) ...
[ERROR 2021-06-25 11:08:33,805] e2e.py: 1268  Command returned non-success status: 1
Traceback (most recent call last):
  File "e2e.py", line 1255, in _run
    kick_off_only=kick_off_only)
  File "e2e.py", line 792, in run_session_command
    f"Command returned non-success status: {status_code}")
RuntimeError: Command returned non-success status: 1
[INFO 2021-06-25 11:08:33,865] e2e.py: 1490  Final check if everything worked.
[INFO 2021-06-25 11:08:33,877] e2e.py: 1496  Final results: {'status': 'timeout', 'last_logs': 'Command returned non-success status: 1'}
Traceback (most recent call last):
  File "e2e.py", line 1675, in <module>
    report=not args.no_report,
  File "e2e.py", line 1599, in run_test
    raise RuntimeError(last_logs)
RuntimeError: Command returned non-success status: 1```

Error output from session
```
Current # nodes: 249, target: 250
Waiting ...
Current # nodes: 249, target: 250
Waiting ...
Traceback (most recent call last):
  File "distributed/test_distributed.py", line 190, in <module>
    test_nodes()
  File "distributed/test_distributed.py", line 38, in test_nodes
    test_max_running_tasks()
  File "distributed/test_distributed.py", line 93, in test_max_running_tasks
    assert ray.get(done[0]) is None
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/client_mode_hook.py", line 47, in wrapper
    return func(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/worker.py", line 1481, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError: ray::task() (pid=13192, ip=172.31.30.0)
  File "python/ray/_raylet.pyx", line 458, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 479, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 349, in ray._raylet.raise_if_dependency_failed
ray.exceptions.RaySystemError: System error: max number of clients reached
traceback: Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/serialization.py", line 246, in deserialize_objects
    obj = self._deserialize_object(data, metadata, object_ref)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/serialization.py", line 197, in _deserialize_object
    return _actor_handle_deserializer(obj)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/serialization.py", line 63, in _actor_handle_deserializer
    serialized_obj, outer_id)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/actor.py", line 960, in _deserialization_helper
    state, outer_object_ref)
  File "python/ray/_raylet.pyx", line 1484, in ray._raylet.CoreWorker.deserialize_and_register_actor_handle
  File "python/ray/_raylet.pyx", line 1453, in ray._raylet.CoreWorker.make_actor_handle
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/function_manager.py", line 400, in load_actor_class
    job_id, actor_creation_function_descriptor)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/function_manager.py", line 486, in _load_actor_class_from_gcs
    ["job_id", "class_name", "module", "class", "actor_method_names"])
  File "/home/ray/anaconda3/lib/python3.7/site-packages/redis/client.py", line 3080, in hmget
    return self.execute_command('HMGET', name, *args)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/redis/client.py", line 898, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/redis/connection.py", line 567, in connect
    self.on_connect()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/redis/connection.py", line 643, in on_connect
    auth_response = self.read_response()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/redis/connection.py", line 739, in read_response
    response = self._parser.read_response()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/redis/connection.py", line 484, in read_response
    raise response
redis.exceptions.ConnectionError: max number of clients reached
```
Link to session:
https://beta.anyscale.com/o/anyscale-internal/projects/prj_6ql9vRLldtxt3yKyLDTdCo/clusters/ses_LfAT4A46D4hacbQUEC1dfTyc

Link to 1.3.0 release issue
https://github.com/ray-project/ray/issues/15214

**Comment by wuisawesome** [comment]
max clients sounds like a configuration issue. strange that it's only coming up on 1.3, but still.

**Comment by DmitriGekhtman** [comment]
hmm, what's our course of action?

**Comment by DmitriGekhtman** [comment]
trying with 1.2.0 just for kicks

**Comment by DmitriGekhtman** [comment]
no luck there:
```
Traceback (most recent call last):
  File "e2e.py", line 1255, in _run
    kick_off_only=kick_off_only)
  File "e2e.py", line 776, in run_session_command
    _check_stop(stop_event)
  File "e2e.py", line 332, in _check_stop
    raise ReleaseTestTimeoutError("Process timed out.")
ReleaseTestTimeoutError: Process timed out.
[WARNING 2021-06-25 13:05:21,372] e2e.py: 1454  Terminating process now.
[INFO 2021-06-25 13:05:21,376] e2e.py: 1490  Final check if everything worked.
[INFO 2021-06-25 13:05:21,379] e2e.py: 1496  Final results: {'status': 'timeout', 'last_logs': 'Process timed out.'}
Traceback (most recent call last):
  File "e2e.py", line 1675, in <module>
    report=not args.no_report,
  File "e2e.py", line 1599, in run_test
    raise RuntimeError(last_logs)
RuntimeError: Process timed out.
```
a bit hard to understand from the command logs what happened in this case
https://beta.anyscale.com/o/anyscale-internal/projects/prj_6ql9vRLldtxt3yKyLDTdCo/clusters/ses_LpPrZRr3A8yNkS796P1ZY9V6?command-history-section=command_history

**Comment by AmeerHajAli** [comment]
Who is working on fixing this?

**Comment by wuisawesome** [comment]
There are a couple of sub issues here:

1. Worker process startup time is extremely high w/ runtime envs (this might be an anyscale product bug, needs confirmation). Mitigating this by turning off runtime envs
2. Fault tolerance issue with actor startup. There are some times where GCS pushes an actor creation task to its assigned worker, but never gets a response when the task finishes. This seems to happen ~0.005% of the time (which causes flakiness since we need to create 10k actors).
3. Task scheduling is _extremely_ slow (causes the envelope to time out). I suspect we can debug this one in the single node envelope (which is also failing).

**Comment by ericl** [comment]
Thanks for digging into this. So what's the solution for (2)? Do we know why the response is lost, is the actor crashing? One solution could be a timeout here.

For (3), is this a problem only for tasks and not actors?

**Comment by wuisawesome** [comment]
> So what's the solution for (2)?

I suspect some sort of timeout + retry is enough. Not seeing any indication from the core worker that it's actually receiving actor creation task.

> For (3), is this a problem only for tasks and not actors?

Yes. Seems to imply some issue with task queuing.

**Comment by ericl** [comment]
How do we distinguish between a timed out or hung rpc and the creation task
just taking a long time?

**Comment by ericl** [comment]
Closing since this is split into separate issues.

**Comment by simon-mo** [comment]
@ericl can you clarify which of the subissues are release blcokers?

**Comment by ericl** [comment]
@simon-mo these two tagged issues:
https://github.com/ray-project/ray/issues/17008
https://github.com/ray-project/ray/issues/17009
